Bookish/Views/Shared/_CopyTable.cshtml

```diff
@@ -7,25 +7,102 @@
             <th scope="col">Id</th>
             <th scope="col">Borrower</th>
             <th scope="col">Due date</th>
+            <th scope="col">Overdue by</th>
+            <th></th>
+            <th></th>
         </tr>
         </thead>
         <tbody>
         @foreach (var copy in Model)
         {
-            <tr>
+            <tr class="@(!copy.IsAvailable() && copy.OverdueTime().Days >= 1 ? "table-danger" : "")">
+                
                 <th scope="row">@copy.Id</th>
+                
                 @if (!copy.IsAvailable())
                 {
                     <td>@copy.Borrower.Name</td>
                     <td>@copy.DueDate</td>
+                    @if (copy.OverdueTime().Days >= 1)
+                    {
+                        <td>@copy.OverdueTime().Days days</td>
+                    }
+                    else
+                    {
+                        <td></td>
+                    }
+                    
+                    <td>
+                        @using (Html.BeginForm("CheckInCopy", "Book", FormMethod.Post))
+                        {
+                            <input type="hidden" id="checkInId" name="Id" value=@copy.Id>
+                            <button type="submit" class="btn btn-primary">Check in</button>
+                        }
+                    </td>
                 }
                 else
                 {
                     <td></td>
                     <td></td>
+                    <td></td>
+                    <td>
+                        <button type="button" class="btn btn-primary" data-bs-toggle="modal" data-bs-target="#checkOutModal" data-copy-id="@copy.Id" onclick="setModalValue(@copy.Id)">Check out</button>
+                    </td>
                 }
+                
+                <td>
+                    @using (Html.BeginForm("DeleteCopy", "Book", FormMethod.Post))
+                    {
+                        <input type="hidden" id="deleteId" name="id" value=@copy.Id>
+                        <button type="submit" class="btn btn-danger">Delete</button>
+                    }
+                </td>
             </tr>
         }
         </tbody>
     </table>
-</div>
\ No newline at end of file
+</div>
+
+<!-- Modal -->

```

Comment:
Nice use of commenting to make it clear what this is - I wonder if this modal block could be separated out into a partial view?